### PR TITLE
Fix HighlightLayer crash when working with an InstancedMesh.

### PR DESCRIPTION
**Pull request by Swiftwork**
Added multiple checks to HighlightLayer to allow InstancedMesh:es. This by referring to their source rather than themselves when adding or removing.

### CODE REVIEW

**Comment by deltakosh** [comment]
Ping @sebavan for validation

**Comment by sebavan** [comment]
All Good on the highlight layer parts, THANK YOU.

**Comment by deltakosh** [comment]
Before I can merge please fix the issue raised by Travis CI

**Comment by sebavan** [comment]
The code now only supports Instances and meshes, no more abastract coud you change it to : 

 if (mesh instanceof InstancedMesh) {
     mesh = (<InstancedMesh> mesh).sourceMesh;
 }

?

**Comment by deltakosh** [comment]
Please test it a bit more deeply as well :)

**Comment by Swiftwork** [comment]
My apologies, it seems as if turned into a bigger issue than first intended.

@sebavan The issue is that an InstancedMesh cannot be type-casted from a Mesh due to them being completely separate classes. This leaves three available options for the methods "mesh" argument:

1. Allow the it to be of either type **Mesh | InstancedMesh**.
2. Use a common ancestor such as **AbstractMesh** (which currently only inherited by Mesh and InstancedMesh).
3. Test for type later and use a wildcard **any** type.

I believe I have seen option number 2 being used previously and therefore assumed that would be the preferred option. Regardless I believe the instanceof checking would look largely similar which ever option is chosen and I am happy to change it for another if you want.

@deltakosh Sure I will try to figure out how to break it ^^. After all the only change I have made to the code is that it refers to an InstancedMesh source rather than itself. I also checked to see if duplicate referrals to the same mesh would cause an issue, but it seemed to be handled fine. If you can come up with a potential issue I could try out I would be happy to :)

Regards
Swiftwork

**Comment by deltakosh** [comment]
No worry, this is fine 👍 
By the way I'm for option 2

**Comment by Swiftwork** [comment]
Any updates on this? Are we waiting for confirmation or validation from some individual? :)

**Comment by deltakosh** [comment]
I'mwauting for your PR update:)

**Comment by Swiftwork** [comment]
Oh sorry, there seems to be some miscommunication. I thought we settled the matter earlier when you said "No worry, this is fine 👍". I have no more commits to add and the latest commit uses the second method posted earlier which everyone seemed content with. I have yet to find any issues, so with your approval and @sebavan I believe it is ready. Travis CI shows no errors now so go ahead and merge when you two OK it! :)

**Comment by deltakosh** [comment]
Oh! Ok:)
@sebavan good to merge?

File: canvas2D/gulp-addModuleExports.js [hunk 1]
```diff
@@ -18,11 +18,16 @@ module.exports = function (varName) {
...
-            'else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;\n' +
-            'return c > 3 && r && Object.defineProperty(target, key, r), r;\n' +
+        'var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {\n' +
+            'var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;\n' +
+            'if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);\n' +
+            'else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;\n' +
+            'return c > 3 && r && Object.defineProperty(target, key, r), r;\n' +
+        '};\n';
+
+        var dependencyAddition =
```
Comment (deltakosh) [inline canvas2D/gulp-addModuleExports.js]: Why this? Seems weird as it will be merge in the babylonjs file

**Comment by sebavan** [comment]
Yep I am good :-) sorry for the delay

File: canvas2D/gulp-addModuleExports.js (hunk 1 above)
Comment (deltakosh) [inline canvas2D/gulp-addModuleExports.js]: Sorry I thought it was the file in bjs gulps folder. Everything's c'est for me

**Pull request merged**